source/behaviors/offClick/offClick.ts

```diff
@@ -29,16 +30,36 @@ export class OffClickDirective implements OnInit, OnDestroy {
 	}
 
 	ngOnInit() {
-		setTimeout(() => {
-			document.addEventListener('click', this.listener);
-		}, 0);
+		if (this.active) {
+			setTimeout(() => {
+				this.addListener();
+			});
+		}
 	}
 
-	ngOnDestroy() {
+	ngOnChanges(changes: SimpleChanges): void {
```

Comment:
The main reason I didn't do that was because I wasn't sure how that would work with setting a default value. I wanted this to default to true so if you didn't set active at all, the directive would still work.